### PR TITLE
leap year added in cpp

### PR DESCRIPTION
**Pull request by tbhaxor**

### CODE REVIEW

File: code/unclassified/leap_year/leap_year.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// part of cosmos by opengenus foundation
+
+// THIS PROGRAM TAKES AN INPUT FOR YEAR AND RETURNS TRUE IF YEAR IS A LEAP YEAR
+
+
+#include <iostream>
+using namespace std;
+
+bool is_leap(int year)
```
Comment (iattempt) [inline code/unclassified/leap_year/leap_year.cpp]: camelCase, please check c++ coding style

File: code/unclassified/leap_year/leap_year.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{
+   return (year%4 == 0 && year%100 != 0) || year%400 == 0;
```
Comment (iattempt) [inline code/unclassified/leap_year/leap_year.cpp]: Please put space around operator and indentation is 4 spaces.
```
year%4 == 0
year % 4 == 0   // valid
```

**Comment by tbhaxor** [comment]
@iattempt i have made some changes plz review them :smile:

File: code/unclassified/leap_year/leap_year.cpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+using namespace std;
+
+bool isLeap(int year)
+{
+   return (year % 4 == 0 && year % 100 != 0) || year % 400 == 0; 
+}
+
+int main()
+{
+   int year = 2000, year_2 = 3000;
```
Comment (arnavb) [inline code/unclassified/leap_year/leap_year.cpp]: The correct name would be year2.

File: code/unclassified/leap_year/leap_year.cpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+   cout<<"Leap years from "<<year<<" to "<<year_2<<endl;
+
+   while(year <= year_2)
+   {
+      if(isLeap(year))
+      {
+         cout<<year<<endl;
```
Comment (arnavb) [inline code/unclassified/leap_year/leap_year.cpp]: Add spaces between all he operators, including <<.

File: code/unclassified/leap_year/leap_year.cpp [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// part of cosmos by opengenus foundation
+
+// THIS PROGRAM TAKES AN INPUT FOR YEAR AND RETURNS TRUE IF YEAR IS A LEAP YEAR
+
+
+#include <iostream>
+using namespace std;
```
Comment (arnavb) [inline code/unclassified/leap_year/leap_year.cpp]: Don’t.

File: code/unclassified/leap_year/leap_year.cpp [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+   cout<<"Leap years from "<<year<<" to "<<year_2<<endl;
+
+   while(year <= year_2)
+   {
+      if(isLeap(year))
+      {
+         cout<<year<<endl;
+      }
+
+      year++;
```
Comment (arnavb) [inline code/unclassified/leap_year/leap_year.cpp]: Pre increment saves copy, although it makes no difference here. Also, why this ruins the point, if I were to get the leap years between two years, I would directly iterate over them, using increments of 4, and checking for 100s at various points. A better way to test his function would be creating tests with catch2, and/or getting a year from user input, and checking if it is a leap year.

**Pull request closed without merging**